Add BartSummarizer using BART transformer for text summarization

kubson2002k opened the pull request:
This PR introduces the `BartSummarizer` class located in `backend/wiki/processors/text/summarizers/bart_summarizer.py`.

Key points:
- Implements abstractive summarization using the `facebook/bart-large-cnn` transformer model via Hugging Face.
- Adds support for summarizing input text up to 1024 characters.
- Returns concise summaries with parameters: `min_length=30`, `max_length=130`.

kubson2002k committed:
Add BartSummarizer using BART transformer for text summarization

kubson2002k committed:
Add BartSummarizer using BART transformer for text summarization

kubson2002k committed:
Add BasicSummarizer that returns top 3 longest sentences as summary

kubson2002k committed:
Change SimilarityPostProcessor to return both labels and texts from process()

kubson2002k committed:
Replace BasicSummarizer with BartSummarizer in BasicPreprocessor for improved summaries

kubson2002k committed:
Extend /clusters endpoint to return article summaries alongside cluster labels

kubson2002k committed:
Add unit tests for BartSummarizer covering short, long, and empty inputs

kubson2002k committed:
Add unit tests for TF-IDF

kubson2002k committed:
Update README with latest project features and checklist

kubson2002k committed:
Update README with summarization module descriptions and unit test overview

kubson2002k committed:
Update README with summarization module descriptions and unit test overview

kubson2002k committed:
Normalize BART summaries: trim whitespace and capitalize first letter

straightchlorine commented on backend/wiki/routes/articles.py:
```diff
@@ -85,11 +85,12 @@ async def cluster_articles_endpoint(
         # attach cluster labels to each article in the result
         for i, article in enumerate(articles):
             article["cluster"] = int(labels[i])
+            article["summarie"]= summaries[i]
```
Here is a typo ☺️ Would be great if you fixed it to `summary` both here and in the docs.

straightchlorine commented on backend/README.md:
```diff
@@ -108,34 +130,37 @@ Querying for the articles and its clusters:
-      "pageid": 46332325,
-      "cluster": 1
+      "cluster": 1,
+      "summarie": "Python high level general purpose programming language design philosophy emphasizes code readability use significant indentation python dynamically type checked garbage collected support multiple programming paradigm including structured particularly procedural object oriented functional programming often described battery included language."
```
And here, in each of the queries ☺️

straightchlorine replied:
Its enough to just fix the line 93 of articles.py and regenerate the JSON for the docs.

straightchlorine reviewed (changes requested):
Would be cool to fix those typos, please ☺️ Should be quick!

kubson2002k committed:
Fix /clusters endpoint to correctly include article summaries in the response

kubson2002k committed:
Add proper summary field to /clusters endpoint response

straightchlorine reviewed (approved):
Thank you very much for contributing ☺️

straightchlorine committed:
Merge branch 'master' into backend-summarization

straightchlorine committed:
fix(tests): adjusted existing tests to the new postprocessor

straightchlorine committed:
fix(tests): adjusted the type hinting and legacy tests

Similar issue - tests were not adjusted to the new format that get
clustering returns.

straightchlorine reviewed (changes requested):
Just as in the comment - fix required for the `TestBasicSummarizerBART`, but also for `TestBasicSummarizerTFIDF`.

straightchlorine committed:
Merge branch 'master' into backend-summarization

kubson2002k committed:
Fix summary length check and empty input test

kubson2002k committed:
Fix summary length check and empty input test

kubson2002k committed:
Fix summary length check and empty input test

kubson2002k committed:
Fix summary length check and empty input test

straightchlorine committed:
Merge branch 'master' into backend-summarization

kubson2002k committed:
Fixed summarizers and tests
